src/T2DMap.cpp

```diff
@@ -3400,6 +3411,20 @@ void T2DMap::slot_setPlayerLocation()
     }
 }
 
+void T2DMap::slot_toggleMapViewOnly()
+{
+    if (mpHost) {
+        mMapViewOnly = !mMapViewOnly;
+        mpHost->mMapViewOnly = mMapViewOnly;
+        TEvent mapModeEvent{};
+        mapModeEvent.mArgumentList.append(QLatin1String("mapModeChangeEvent"));
+        mapModeEvent.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+        mapModeEvent.mArgumentList.append(QLatin1String(mMapViewOnly ? "viewing" : "editing"));
```

Comment:
Just to be safe I'd do

```suggestion
        mapModeEvent.mArgumentList.append(mMapViewOnly ? QStringLiteral("viewing") : QStringLiteral("editing")));
```

`QStringLiteral` is a cool thing that generates the QStrings at compile-time for us, making them essentially free at runtime.